### ISSUE TITLE
Support JFR Event Streaming

### ISSUE THREAD

**Issue by roberttoyonaga** [oracle/graal#5079]
### TL;DR
Add support for JFR event streaming to native images. 

### Goals
Implement functionality required to support the JFR event streaming feature. Users should be able to use the JFR streaming API as they normally would in their application code. Streaming from a running recording or from file.

### Details
Streaming of JFR events is possible since jdk14. This allows users to take advantage of the the JFR streaming API to gain access to event data without having to dump a JFR snapshot. Currently the methods required to support streaming are substituted and unimplemented in substrate VM. In order to stream from an ongoing recording chunks must be able to be flushed without actually being rotated. In hotspot, the flushing thread and the thread that monitors the JFR disk repo for events to stream are excluded from JFR sampling.  Something similar should be done in substrateVM.
